source/extensions/http/stateful_session/cookie/cookie.cc

```diff
@@ -11,18 +12,33 @@ namespace Cookie {
 void CookieBasedSessionStateFactory::SessionStateImpl::onUpdate(
     const Upstream::HostDescription& host, Envoy::Http::ResponseHeaderMap& headers) {
   absl::string_view host_address = host.address()->asStringView();
+  std::string encoded_address;
   if (!upstream_address_.has_value() || host_address != upstream_address_.value()) {
-    const std::string encoded_address =
-        Envoy::Base64::encode(host_address.data(), host_address.length());
+    if (Runtime::runtimeFeatureEnabled(
+            "envoy.reloadable_features.stateful_session_encode_ttl_in_cookie") &&
+        !use_old_style_encoding_) {
+      auto expiry_time = std::chrono::duration_cast<std::chrono::seconds>(
+          (time_source_.monotonicTime() + std::chrono::seconds(factory_.ttl_)).time_since_epoch());
+      // Build proto message
+      envoy::Cookie cookie;
+      cookie.set_address(std::string(host_address));
+      cookie.set_expires(expiry_time.count());
+      std::string proto_string;
+      cookie.SerializeToString(&proto_string);
+
+      encoded_address = Envoy::Base64::encode(proto_string.data(), proto_string.length());
```

Comment:
Could we use base64 to encode the proto binary rather than serialized string?

Reply:
I think serialization is necessary. the `string` in `SerializeToString` is a binary string (pointer and length). It is the wire format of the protobuf. Please see https://protobuf.dev/reference/cpp/cpp-generated (look for SerializeToString). The thing is that when the cookie is received, the code must be resilient against interpreting the payload in wrong way, which would result in a crash. protobuf's ParseFromString does safe de-serialization from wire format.

Reply:
Got it. I misunderstood the meaning for this method. Then this is what I want.